scripts/xsslint/xsslint/main.py

```diff
@@ -102,6 +105,107 @@ def _process_os_dirs(starting_dir, template_linters, options, summary_results, o
 
 
+def _get_xsslint_counts(result_contents):
+    """
+    This returns a dict of violations from the xsslint report.
+
+    Arguments:
+        filename: The name of the xsslint report.
+
+    Returns:
+        A dict containing the following:
+            rules: A dict containing the count for each rule as follows:
+                violation-rule-id: N, where N is the number of violations
+            total: M, where M is the number of total violations
+
+    """
+
+    rule_count_regex = re.compile(r"^(?P<rule_id>[a-z-]+):\s+(?P<count>\d+) violations", re.MULTILINE)
+    total_count_regex = re.compile(r"^(?P<count>\d+) violations total", re.MULTILINE)
+    violations = {'rules': {}}
+    for violation_match in rule_count_regex.finditer(result_contents):
+        try:
+            violations['rules'][violation_match.group('rule_id')] = int(violation_match.group('count'))
+        except ValueError:
+            violations['rules'][violation_match.group('rule_id')] = None
+    try:
+        violations['total'] = int(total_count_regex.search(result_contents).group('count'))
+    # An AttributeError will occur if the regex finds no matches.
+    # A ValueError will occur if the returned regex cannot be cast as a float.
+    except (AttributeError, ValueError):
+        violations['total'] = None
+    return violations
+
+
+def _check_violations(options, results):
+    try:
+        thresholds_option = options['thresholds']
+        # Read the JSON file
+        with open(thresholds_option, 'r') as file:
+            violation_thresholds = json.load(file)
+
+    except ValueError:
+        violation_thresholds = None
+    if isinstance(violation_thresholds, dict) is False or \
+            any(key not in ("total", "rules") for key in violation_thresholds.keys()):
+        print('xsslint')
+        print("""FAILURE: Thresholds option "{thresholds_option}" was not supplied using proper format.\n"""
+              """Here is a properly formatted example, '{{"total":100,"rules":{{"javascript-escape":0}}}}' """
+              """with property names in double-quotes.""".format(thresholds_option=thresholds_option))
+    xsslint_script = "xss_linter.py"
+
+    try:
+        metrics_str = "Number of {xsslint_script} violations: {num_violations}\n".format(
+            xsslint_script=xsslint_script, num_violations=int(results['total'])
+        )
+        if 'rules' in results and any(results['rules']):
+            metrics_str += "\n"
+            rule_keys = sorted(results['rules'].keys())
+            for rule in rule_keys:
+                metrics_str += "{rule} violations: {count}\n".format(
+                    rule=rule,
+                    count=int(results['rules'][rule])
+                )
+    except TypeError:
+        print('xsslint')
+        print("FAILURE: Number of {xsslint_script} violations could not be found".format(
+            xsslint_script=xsslint_script
+        ))
```

Comment:
@salman2013 , please think about what would happen in the check if `int(results['total'])` raised a TypeError. Would the check pass or fail? Would we want the check to pass or fail? Let me know if my questions do not make sense.

I would ask the same questions about every occurance of `print("FAILURE: ...")` in this script.

Reply:
Yes, i got your point, i have fixed this.